### DISCUSSION TITLE
Option for tscn files to have a designated child destination node

### DISCUSSION THREAD

**Discussion by yahkr** [godotengine/godot-proposals#8201] [GUI]
While attempting to create a generic Window for my game's ui I ran into this puzzle. If I have the following scene:
![image](https://github.com/godotengine/godot-proposals/assets/62478788/d71fb31a-9e95-4dc9-8472-3c57a2b9927e)
and add a button to it this is what happens:
![image](https://github.com/godotengine/godot-proposals/assets/62478788/ef7df059-6dfb-4545-a479-b67aa48524e8)
This is what I would expect to happen:
![image](https://github.com/godotengine/godot-proposals/assets/62478788/20149c5d-ad21-487f-9505-104640bf7499)
And this works with editable children, however if I now modify the scene file in some way that moves the ContentContainer child, all hell breaks loose and the button is no longer a child of the ContentContainer and becomes named its old location and is not deletable until I restart the entire editor.
![image](https://github.com/godotengine/godot-proposals/assets/62478788/0ea8581d-2805-4fc0-a3a8-01c69bca606a)

---------------------------------------------
This is how I would hope it would work. Being able to add/remove children without worry about the underlying scene changing, all they care about is where do I end up when added as a child. 
![image](https://github.com/godotengine/godot-proposals/assets/62478788/002dec37-3415-4cc3-874e-fb7b293aea3d)

I did attempt to create a custom node with the following basic code, but with this solution the children are inaccessible.
```cs
[Tool]
public partial class DraggableWindow : VBoxContainer
{
    private Node root;
    private Node contentContainer;
    
    public override void _Ready()
    {
        var scene = GD.Load<PackedScene>("res://addons/JUI/Window.tscn");
        root = scene.Instantiate();
        contentContainer = root.GetNode<VBoxContainer>("%ContentContainer");
        AddChild(root);
    }
    public override void _Notification(int what)
    {
        if (what == NotificationChildOrderChanged)
        {
            if(GetChildren().Count > 1)
            {
                GetChild<Node>(1).Reparent(contentContainer);
            }
        }
    }
}
```
![image](https://github.com/godotengine/godot-proposals/assets/62478788/f395b5a9-c097-4ee2-9d35-985192cb7c5f)

**Comment by theraot** [comment]
See https://github.com/godotengine/godot-proposals/issues/8184
